File: cosmol_viewer/examples/render_molecule.py
from cosmol_viewer import Molecules, Scene, Viewer

mol_data = open("./examples/6fi1_ligand.sdf", "r", encoding="utf-8").read()

mol = Molecules.from_sdf(mol_data).centered()

scene = Scene()

scene.set_scale(1.0)

scene.add_shape_with_id("molecule", mol)

viewer = Viewer.render(scene, width=800, height=500)

viewer.save_image("screenshot.png")

print("Press Any Key to exit...", end="", flush=True)
_ = input()

File: cosmol_viewer/examples/render_molecule.rs
use cosmol_viewer::parser::sdf::{ParserOptions, parse_sdf};
use cosmol_viewer::{Scene, Viewer, shapes::Molecules};
use std::path::Path;

fn main() {
    let sdf_string = std::fs::read_to_string("./examples/example.sdf").unwrap();
    let opts = ParserOptions {
        keep_h: true,
        multimodel: true,
        onemol: false,
    };
    let mol_data = parse_sdf(&sdf_string, &opts);
    let mol = Molecules::new(mol_data).centered();

    let mut scene = Scene::new();

    scene.add_shape(mol, Some("mol"));

    let viewer = Viewer::render(&scene, 800.0, 500.0);

    let img = viewer.take_screenshot();

    img.save(Path::new("screenshot.png")).unwrap();

    println!("Press Enter to exit...");
    use std::io::{self, Write};
    let _ = io::stdout().flush();
    let _ = io::stdin().read_line(&mut String::new());
}
